src/code/fv-components/end-level-dialog.js

```diff
@@ -0,0 +1,79 @@
+import React, {PropTypes} from 'react';
+import t from '../utilities/translate';
+import GemSetView, {GemView} from './gem-set';
+import { getChallengeScores } from '../reducers/helpers/challenge-progress';
+
+const EndLevelDialogView = ({challengeAwards, onNextChallenge, onTryAgain}) => {
+  let getGemDisplayName = (score) => {
+    let gemName = t("~VENTURE.AMBER_GEM");
+    if (score === 1) gemName = t("~VENTURE.BLUE_GEM");
+    if (score >= 2) gemName = t("~VENTURE.RED_GEM");
+    return gemName;
+  };
+
+  if (challengeAwards.routeSpec == null) {
+    return null;
+  }
+
+  let level = challengeAwards.routeSpec.level,
+      mission = challengeAwards.routeSpec.mission,
+      challenge = challengeAwards.routeSpec.challenge,
+      challengeCount = challengeAwards.challengeCount,
+      progress = challengeAwards.progress;
+
+  let challengeScores = getChallengeScores(level, mission, challengeCount, progress),
+      currentScore = challengeScores[challenge];
+  return (
+    <div className="end-level-dialog-container">
+      <div className="end-level-dialog-backdrop"></div>
+      <div className="end-level-dialog-background"></div>
+      <div className="end-level-dialog-overlay"></div>
+      <div className="end-level-dialog-screen">
+        <div className="challenge-complete-text">
+          /{t("~VENTURE.END_LEVEL")}
+        </div>
+        <div className="end-level-flavor-text">
+          {"Geni_retrieve <get> boot_all </get>"}
```

Comment:
Lots of "flavor text" I took out of FV screenshots here...I figured it's mostly gibberish so it wouldn't need to be localized later.